cycledash/static/js/runs/components/RunsPage.js

```diff
@@ -179,14 +179,18 @@ var ProjectTable = React.createClass({
     var numBams = this.props.bams.length;
     var numRuns = this.props.runs.length;
     var table;
-    if (this.state.bamsTable) {
+    if (this.state.bamsTable && numBams > 0) {
       table = <BamsTable bams={this.props.bams}
                          selectedBamId={this.state.selectedBamId}
                          createClickBamHandler={this.createClickBamHandler} />;
-    } else {
+    } else if (!this.state.bamsTable && numRuns > 0) {
       table = <RunsTable runs={this.props.runs}
                          selectedRunId={this.state.selectedRunId}
                          createClickRunHandler={this.createClickRunHandler} />;
+    } else if (this.state.bamsTable && numBams === 0) {
```

Comment:
These two cases (when numBams or rows is 0) can be handled in the RunsTable and BamsTable components instead; that way you avoid a big if/else block here.